File: Leetcode/1518.cs
namespace Leetcode
{
    internal class _1518
    {
        public int NumWaterBottles(int numBottles, int numExchange)
        {
            if (numBottles < numExchange)
            {
                return numBottles;
            }

            int emptyBottles = numBottles,
                res = numBottles;

            while (emptyBottles >= numExchange)
            {
                var newFullBottles = emptyBottles / numExchange;
                res += newFullBottles;
                var newEmptyBottles = newFullBottles;
                emptyBottles = (emptyBottles % numExchange) + newEmptyBottles;
            }

            return res;
        }
    }
}

File: Leetcode/go/1518.go
package main

func numWaterBottles(numBottles int, numExchange int) int {
	if numBottles < numExchange {
		return numBottles
	}

	res, emptyBottles := numBottles, numBottles
	for emptyBottles >= numExchange {
		newFullBottles := emptyBottles / numExchange
		res += newFullBottles

		newEmptyBottles := newFullBottles
		emptyBottles = emptyBottles%numExchange + newEmptyBottles
	}
	return res
}
